packages/storage/src/reference.ts

```diff
@@ -173,7 +173,7 @@ export class Reference {
     );
     this.throwIfRoot_('putString');
     let data = fbsString.dataFromString(format, string);
-    let metadata = object.clone<Metadata>(opt_metadata);
+    let metadata = clone(opt_metadata);
```

Comment:
Consider `Object.assign({}, opt_metadata)`

Reply:
Done in next PR.